lib/batch_loader.rb

```diff
@@ -14,18 +14,20 @@ class BatchLoader
 
   NoBatchError = Class.new(StandardError)
 
-  def self.for(item)
-    new(item: item)
+  def self.for(item, slug = nil, context = nil)
+    new(item: item, slug: slug, context: context)
   end
 
-  def initialize(item:)
+  def initialize(item:, slug:, context:)
```

Comment:
Having a `context` is a nice idea but at this point, I'd prefer making as few changes as possible to the existing API :)

For example, adding just a new optional `key` argument:

```ruby
BatchLoader.for(id).batch(key: "mykey") do |ids, loader|
  ...
end
```

There are a few options how to name the argument: `slug`, `id`, `key`. Maybe it makes sense to name it `key` similarly to [React](https://reactjs.org/docs/lists-and-keys.html#keys) where it helps to solve a similar problem of identifying the element. Additionally, the options `cache` and `key` fit nicely together in case if `BatchLoader` will have a more powerful caching mechanism in the future (in memory, LRU, TTL, external storages, etc.). 

Please let me know what do you think about the naming?

Reply:
Must agree I thought slug was an odd choice, I think `key` is more intuitive.

Reply:
Good idea! I agree and i will be do it... Tks =]